src/pdsfile/pdscache.py

```
@@ -126,19 +247,47 @@ def resume(self):
+        read is still reported as present. Reading it through ``get()`` removes it, and
+        the same test then answers False.
+
+        Parameters:
+            key: the key to test.
+
+        Returns:
+            bool: True if the key has an entry.
+        """
         return (key in self.dict)
 
     def __len__(self):
```

Ruff (0.16.1)

[warning] Missing return type annotation for special method `__len__`

Add return type annotation: `int`

(ANN204)